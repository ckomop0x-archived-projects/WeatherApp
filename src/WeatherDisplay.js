import React, { Component } from 'react';

class WeatherDisplay extends Component {
	constructor() {
		super();
		this.state = {
			weatherData: null
		};
	}
	componentDidMount() {
		const zip = this.props.zip;
		const URL =`https://crossorigin.me/http://api.openweathermap.org/data/2.5/weather?q="${zip}&appid=b1b35bba8b434a28a0be2a3e1071ae5b&units=imperial`;
		fetch(URL)
			.then(res => res.json())
			.then(json => {
				this.setState({ weatherData: json });
		});
	}
	render() {
		const weatherData = this.state.weatherData;
		if ( !weatherData ) {
			return <div>Loading</div>;
		} else { const weather = weatherData.weather[0];
			const iconUrl = "http://openweathermap.org/img/w/" + weather.icon + ".png";
			return (
				<div>
					<h1>
						{weather.main} in {weatherData.name}
						<img src={iconUrl} alt={weatherData.description} />
					</h1>
					<p>Current: {weatherData.main.temp}°</p>
					<p>High: {weatherData.main.temp_max}°</p>
					<p>Low: {weatherData.main.temp_min}°</p>
					<p>Wind Speed: {weatherData.wind.speed} mi/hr</p>
				</div>
			);
		}
	}
}

export default WeatherDisplay;