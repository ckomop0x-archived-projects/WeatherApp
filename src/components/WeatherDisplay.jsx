import React, {Component} from 'react';
import {urlMaker} from '../services/url-maker';

class WeatherDisplay extends Component {
    constructor (props) {
        super(props);
        this.state = {
            weatherData: null,
            city: this.props.city || 'amsterdam'
        };
    }

    componentDidMount () {
        this.loadWeather(this.props.city);
    }

    loadWeather (city) {
        fetch(urlMaker(city))
            .then(res => res.json())
            .then(json => {
                this.setState({weatherData: json});
            });
    }

    componentWillReceiveProps (newProps) {
        if (newProps.city !== this.state.city) {
            this.setState({city: newProps.city})
            this.loadWeather(newProps.city);
        }
    }

    render () {
        const {weatherData} = this.state;

        if (!weatherData) {
            return (
                <div>Loading<sup>*</sup></div>
            );
        } else {
            const weather = weatherData.weather[0];
            const iconUrl = 'http://openweathermap.org/img/w/' + weather.icon + '.png';

            return (
                <div>
                    <div className="card" style={{width: 400, marginTop: 32}}>
                        {/*<img className="card-img-top" src="https://funtime.kiev.ua/uploads/img/gallery/big/2016/10/uzhgorod-uzhgorod-ukraina-uzhgorodskij-zamok-uzhgorodskaya-oblast-uzhgorod-dostoprimechatelnosti-uzh-57fb1d4ceb34d.jpg"*/}
                             {/*alt={weatherData.name} />*/}
                            <div className="card-body">
                                <h5 className="card-title">{weatherData.name}</h5>
                                <p className="card-text">{weather.main} <img src={iconUrl} alt={weatherData.description}/></p>
                            </div>
                            <ul className="list-group list-group-flush">
                                <li className="list-group-item">Current: {weatherData.main.temp}°</li>
                                <li className="list-group-item">High: {weatherData.main.temp_max}°</li>
                                <li className="list-group-item">Low: {weatherData.main.temp_min}°</li>
                                <li className="list-group-item">Wind Speed: {weatherData.wind.speed} mi/hr</li>
                            </ul>
                    </div>
                </div>
            );
        }
    }
}

export default WeatherDisplay;
