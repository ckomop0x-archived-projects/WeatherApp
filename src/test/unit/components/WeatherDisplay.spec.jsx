import React from 'react';
import renderer from 'react-test-renderer';
import WeatherDisplay from '../../../components/WeatherDisplay';


describe('Components', () => {
    describe('App', () => {
        it('renders without crashing', () => {
            const wrapper = renderer.create(<WeatherDisplay/>).toJSON();
            expect(wrapper).toMatchSnapshot();
        });
    });
});