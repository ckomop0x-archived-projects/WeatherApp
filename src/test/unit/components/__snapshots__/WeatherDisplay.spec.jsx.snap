// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components App renders without crashing 1`] = `
<div>
  Loading
  <sup>
    *
  </sup>
</div>
`;
